pkg/providers/launchtemplate/suite_test.go

```diff
@@ -1560,7 +1560,7 @@ var _ = Describe("LaunchTemplates", func() {
 				Expect(*input.LaunchTemplateData.ImageId).To(ContainSubstring("test-ami"))
 			})
 		})
-		Context("Subnet-based Launch Template Configration", func() {
+		Context("Public IP Association", func() {
```

Comment:
Should we add or change our E2E tests?

Reply:
So these changes are a little difficult to test in E2E and I'm not sure if there's much advantage to increasing the scope of the existing tests. As long as the lt is correctly generated I'm not worried EFA vs non-EFA instances from an E2E standpoint.

Reply:
E.G. on the difficult to test point, can't test setting the association to false in a public subnet. There's probably some work we could do by grabbing the provider ID off the nodeclass and describing the instance through the AWS API but I don't think that really gains us anything over the existing tests. The LT tests plus a couple smoke tests seem sufficient to me.

Reply:
Do we do any specific E2E testing for EFA?

Reply:
Ya we do, in the extended resources tests: https://github.com/aws/karpenter-provider-aws/blob/d9a2d1b95ce39856d93ea68600a38d468333b8c4/test/suites/integration/extended_resources_test.go#L214